### PR TITLE
Split TypeScript assignment into TSX and TypeScript

### PR DESCRIPTION
**Pull request by joshvera**
In https://github.com/tree-sitter/tree-sitter-typescript/pull/68, we separated the TSX and TypeScript syntax in order to support type arguments in TSX and avoid ambiguity.

This PR adds a new assignment to semantic and moves `JSX`, `TSX`, and `JavaScript` to this new parser/assignment in order to assign more precisely. In practice this means type assertions are no longer supported in Flow/JS and JSX fragments are no longer parsed in TypeScript.

### CODE REVIEW

File: src/Language/TSX/Assignment.hs [hunk 1]
```diff
@@ -0,0 +1,971 @@
...
+  where
+    everything = choice [
+      asExpression,
+      nonNullExpression',
+      importAlias',
+      internalModule,
+      super,
+      object,
+      array,
+      jsxElement',
```
Comment (joshvera) [inline src/Language/TSX/Assignment.hs]: I've had to duplicate typescript assignment in order to leave `jsxElement'` and `jsxFragment` in here. I could move assignment out into some `Common` module and pass in a list of terms to mappend instead. Thoughts?

Comment (rewinfrey) [reply src/Language/TSX/Assignment.hs]: Thanks for calling out the duplication. Given the size of this patch as it is, I'd be in favor of merging this and to extract a common assignment module as a follow up PR. That might also make a weightier issue for someone in the community to take on. Either way I'm 👍

File: src/Language/TSX/Resolution.hs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                  <> maybe mempty (:[]) packageDotJSON
+                  <> (((path </> "index") <.>) <$> exts)
+  trace ("searching in " <> show searchPaths)
+  maybe (Left searchPaths) Right <$> resolve searchPaths
+
+typescriptExtensions :: [String]
+typescriptExtensions = ["ts", "tsx", "d.ts"]
+
+javascriptExtensions :: [String]
+javascriptExtensions = ["js"]
```
Comment (rewinfrey) [inline src/Language/TSX/Resolution.hs]: Are there differences between this `Resolution` module and `TypeScript.Resolution`? I'm curious if we can still maintain a single `Resolution` module for `TypeScript` and `TSX`?

File: test/Examples.hs [hunk 3]
```diff
@@ -83,7 +83,8 @@ languages :: [LanguageExample]
 languages =
   [ le "python" ".py" "examples" (Just "script/known_failures.txt")
   , le "ruby" ".rb" "examples" (Just "script/known_failures.txt")
-  , le "typescript" ".ts" "examples" (Just "script/known_failures.txt")
+  , le "typescript" ".ts" "examples" (Just "typescript/script/known_failures.txt")
+  , le "typescript" ".tsx" "examples" (Just "typescript/script/known_failures.txt")
```
Comment (rewinfrey) [inline test/Examples.hs]: I'm not familiar with this context, but is `"typescript"` as the language correct here rather than `"tsx"`?

Comment (joshvera) [reply test/Examples.hs]: Yeah `"typescript"` refers to the directory the example repos are in. This adds about 150 tests to `parse-examples`.

**Pull request merged**